### PR TITLE
chore: Update yarn.lock to remove inflight from application dependency tree

### PR DESCRIPTION
**Pull request by danjm**
## **Description**

We have a `yarn audit` failure on develop due to the package `inflight` which is used by the package `glob`.  While `glob` is a dependency of many of our _dev_ dependencies, it is only causing an audit failure because it is a dependency of a package ultimately imported by trezor:

```
danjm@pop-os:~/metamask/metamask-extension$ yarn why -R inflight
└─ metamask-crx@workspace:.
   ├─ @metamask/eth-trezor-keyring@npm:3.1.0 (via npm:^3.1.0)
   │  └─ @trezor/connect-web@npm:9.2.2 [42492] (via npm:^9.1.11 [42492])
   │     ├─ @trezor/connect@npm:9.2.2 [d8f53] (via npm:9.2.2 [d8f53])
   │     │  ├─ @trezor/connect-analytics@npm:1.0.14 [bb3a7] (via npm:1.0.14 [bb3a7])
   │     │  │  └─ @trezor/analytics@npm:1.0.16 [bc8f6] (via npm:1.0.16 [bc8f6])
   │     │  │     └─ @trezor/env-utils@npm:1.0.15 [342f9] (via npm:1.0.15 [342f9])
   │     │  │        └─ expo-constants@npm:15.4.5 [20ef0] (via npm:15.4.5 [20ef0])
   │     │  │           └─ @expo/config@npm:8.5.6 (via npm:~8.5.0)
   │     │  │              ├─ @expo/config-plugins@npm:7.9.2 (via npm:~7.9.0)
   │     │  │              │  └─ glob@npm:7.1.6 (via npm:7.1.6)
   │     │  │              │     └─ inflight@npm:1.0.6 (via npm:^1.0.4)
   │     │  │              ├─ glob@npm:7.1.6 (via npm:7.1.6)
   │     │  │              └─ sucrase@npm:3.34.0 (via npm:3.34.0)
   │     │  │                 └─ glob@npm:7.1.6 (via npm:7.1.6)
```
However, `expo-constants` is not actually used in our application code. It's inclusion in our dependency tree is due to some problem in how the trezor packages are built (which we also should probably investigate, but that can be done separately).

~~To solve the audit failure, this PR edits yarn.lock to resolve the instances of `glob` under `@expo/config` to a version that does not use `inflight`.~~

Update: as per the below suggestion, instead of directly editing the yarn.lock file, this PR now adds resolutions to resolve the instances of `glob` under `@expo/config` to a version that does not use `inflight`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24737?quickstart=1)

## **Manual testing steps**

yarn audit should pass

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by legobeat** [comment]
@danjm The canonical way to address this would be through `resolutions` in `package.json` to avoid inadvertent revert of these down the line.

The following lines should do it:
```
    "sucrase@npm:3.34.0": "^3.35.0",
    "@expo/config/glob": "^9",
    "@expo/config-plugins/glob": "^9",
```

**Comment by danjm** [comment]
Thanks @legobeat, I can update to match your suggestions

File: yarn.lock [hunk 1]
```diff
@@ -2889,7 +2889,7 @@ __metadata:
     debug: "npm:^4.3.1"
     find-up: "npm:~5.0.0"
     getenv: "npm:^1.0.0"
-    glob: "npm:7.1.6"
+    glob: "npm:^9.0.0"
```
Comment (legobeat) [inline yarn.lock:2892]:
```suggestion
    glob: "npm:7.1.6"
```

File: yarn.lock [hunk 2]
```diff
@@ -2916,12 +2916,12 @@ __metadata:
     "@expo/config-types": "npm:^50.0.0"
     "@expo/json-file": "npm:^8.2.37"
     getenv: "npm:^1.0.0"
-    glob: "npm:7.1.6"
+    glob: "npm:^9.0.0"
```
Comment (legobeat) [inline yarn.lock:2919]:
```suggestion
    glob: "npm:7.1.6"
```

File: yarn.lock [hunk 3]
```diff
@@ -2916,12 +2916,12 @@ __metadata:
...
     require-from-string: "npm:^2.0.2"
     resolve-from: "npm:^5.0.0"
     semver: "npm:7.5.3"
     slugify: "npm:^1.3.4"
-    sucrase: "npm:3.34.0"
+    sucrase: "npm:^3.35.0"
```
Comment (legobeat) [inline yarn.lock:2924]:
```suggestion
    sucrase: "npm:3.34.0"
```

**Review by DDDDDanica — DISMISSED** [review]

File: yarn.lock (hunk 1 above)
Comment (danjm) [reply yarn.lock:2892]: For some reason github won't let me commit this here, saying "the diff has recently been update". I will push a commit that makes these changes.

Comment (danjm) [reply yarn.lock]: 012944bf4e

File: yarn.lock (hunk 2 above)
Comment (danjm) [reply yarn.lock]: 012944bf4e

File: yarn.lock (hunk 3 above)
Comment (danjm) [reply yarn.lock]: 012944bf4e

**Review by legobeat — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**